pkg/controller/registry/resolver/operators.go

```diff
@@ -266,6 +265,34 @@ func NewOperatorFromBundle(bundle *api.Bundle, replaces string, startingCSV stri
 	for _, gvk := range bundle.RequiredApis {
 		required[registry.APIKey{Plural: gvk.Plural, Group: gvk.Group, Kind: gvk.Kind, Version: gvk.Version}] = struct{}{}
 	}
+	sourceInfo := &OperatorSourceInfo{
+		Package:     bundle.PackageName,
+		Channel:     bundle.ChannelName,
+		StartingCSV: startingCSV,
+		Catalog:     sourceKey,
+	}
+
+	// legacy support - if the grpc api doesn't contain the information we need, fallback to csv parsing
+	if len(required) == 0 && len(provided) == 0 {
```

Comment:
Just a thought -- but if JSONPath can let us query the provided and required APIs w/o unmarshaling the entire CSV, should we use it to more accurately verify the content of the bundle fields? 

(this question shouldn't block the PR)

Reply:
I was thinking that if the grpc has any provide/required apis, we just trust that (gives us more freedom on the side of the registry going forward).